wemake_python_styleguide/logic/tree/imports.py

```diff
@@ -35,3 +36,14 @@ def is_vague_import(name: str) -> bool:
     )
     too_short = logical.is_too_short_name(name, 2, trim=True)
     return blacklisted or with_from_or_to or too_short
+
+
+def is_nested_typing_import(parent: ast.AST) -> bool:
+    """Tells whether ``if`` checks for ``TYPE_CHECKING``."""
+    checked_condition = None
+    if isinstance(parent, ast.If):
+        if isinstance(parent.test, ast.Name):
+            checked_condition = parent.test.id
+        elif isinstance(parent.test, ast.Attribute):
+            checked_condition = parent.test.attr
+    return checked_condition == 'TYPE_CHECKING'
```

Comment:
Let's move it to constants.

Reply:
I've added it as a `frozenset`, since it looked a bit more natural to me than a single name, but if you'd prefer
```
#: Allows to have imports inside ``TYPE_CHECKING``
ALLOWED_NESTED_IMPORTS_CONDITION: Final = 'TYPE_CHECKING'
```
or a variation, I can change